src/ray/gcs/gcs_server/gcs_health_check_manager.h

```diff
@@ -16,16 +16,19 @@
 
 #include <grpcpp/grpcpp.h>
 
+#include <cstdint>
+#include <functional>
+#include <memory>
+#include <vector>
+
 #include "absl/container/flat_hash_map.h"
 #include "ray/common/asio/instrumented_io_context.h"
 #include "ray/common/id.h"
 #include "ray/common/ray_config.h"
+#include "ray/util/thread_checker.h"
 #include "src/proto/grpc/health/v1/health.grpc.pb.h"
 
-class GcsHealthCheckManagerTest;
-
-namespace ray {
-namespace gcs {
+namespace ray::gcs {
 
 /// GcsHealthCheckManager is used to track the healthiness of the nodes in the ray
```

Comment:
Add comment about all public methods should be called on the same thread, that is the thread io_service_ lives on.

Reply:
Add comment on IO operation threading.